tradeexecutor/analysis/trade_analyser.py

```diff
@@ -576,15 +577,24 @@ def show_custom(df: pd.DataFrame):
         with pd.option_context("display.max_row", None):
             display(df.style.set_table_styles([{'selector': 'thead', 'props': [('display', 'none')]}]))
     
-    def check_quantstats(self):
-        """Check that requirements are met for using quantstats library. Used as decorator for provided helper methods."""
-        if not HAS_QUANTSTATS:
-            raise RuntimeError("Quantstats library not installed")
-
-        if not hasattr(self, "daily_returns"):
-            raise RuntimeError("Daily returns have not been calculated. Remember to provided state \
-                argument. E.g. summary = analysis.calculate_summary_statistics(state=state)")
-
+    @staticmethod
+    def check_quantstats(function):
+        """Decorator function that checks that requirements are met for using quantstats library. Used as decorator for provided helper methods."""
+        def wrapper(*args, **kwargs):
+            if not HAS_QUANTSTATS:
+                raise RuntimeError("Quantstats library not installed")
+
+            if not hasattr(args[0], "daily_returns"):
+                raise RuntimeError("Daily returns have not been calculated. Remember to provided state argument. E.g. summary = analysis.calculate_summary_statistics(state=state)")
+            
+            with warnings.catch_warnings():
```

Comment:
Can you add a comment what warnings have been silenced

(Copy paste warning message)